File: 2929_Distribute_Candies_Among_Children_II.cpp
class Solution {
public:
    long long distributeCandies(int n, int limit) {
        long long yb = 0;
        for (int x = 0; x <= limit; ++x) {
            int rem = n - x;
            if (rem < 0) break;
            int min_y = std::max(0, rem - limit);
            int max_y = std::min(limit, rem);
            if (max_y >= min_y) {
                yb += (max_y - min_y + 1);
            }
        }
        return yb;
    }
};

File: 2929_Distribute_Candies_Among_Children_II.py
class Solution(object):
    def distributeCandies(self, n, limit):
        """
        :type n: int
        :type limit: int
        :rtype: int
        """
        yb = 0
        for x in range(limit + 1):
            rem = n - x
            if rem < 0:
                break
            min_y = max(0, rem - limit)
            max_y = min(limit, rem)
            if max_y >= min_y:
                yb += (max_y - min_y + 1)
        return yb
